src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -72,13 +66,14 @@ protected override int GetFields<T>(T instance, LazyStringValue key, object docu
                 object value;
                 if (obj.IsObject() && obj.IsArray() == false)
                 {
-                    //Note that for dynamic fields 'field.Name' will contain the real name of the field and not 'property'
+                    //In case TryDetectDynamicFieldCreation finds a dynamic field it will populate 'field.Name' witht the actual property name 
```

Comment:
`witht`

Reply:
Will this mutate the actual `field.Name` instance?

Reply:
yes, this is why we pass the field to the method.

Reply:
see https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs#L137

Reply:
Isn't this the field that we just pulled from the cache, so we are mutating a global value here?

Reply:
it shouldn't because the field name is originaly '_' and we modify it to something else. but I'll need to verify this.